etc/fish_completion/kw.fish

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env fish
+
+# Test if the kw command has subcommands
+function __fish_kw_no_commands
+    set cmd (commandline -opc)
+    if [ "$cmd" = "kw" ]
+    	return 0
+    end
+
+    return 1
+end
+
+# Disable file completion when there's no specified command
+complete -c kw -n __fish_kw_no_commands -f
+
+# kw commands
+complete -c kw -n "__fish_kw_no_commands" -a "build b" -d "Build Kernel and modules"
+complete -c kw -n "__fish_kw_no_commands" -a "install i" -d "Install modules"
+complete -c kw -n "__fish_kw_no_commands" -a "bi" -d "Build and install modules"
+complete -c kw -n "__fish_kw_no_commands" -a "prepare p" -d "Deploy basic environment in the VM"
+complete -c kw -n "__fish_kw_no_commands" -a "new n" -d "Install new Kernel image"
+complete -c kw -n "__fish_kw_no_commands" -a "ssh s" -d "Enter in the vm"
+complete -c kw -n "__fish_kw_no_commands" -a "mount mo" -d "Mount partition with qemu-nbd"
+complete -c kw -n "__fish_kw_no_commands" -a "umount um" -d "Umount partition created with qemu-nbd"
+complete -c kw -n "__fish_kw_no_commands" -a "vars v" -d "Show variables"
+complete -c kw -n "__fish_kw_no_commands" -a "up u" -d "Wake up vm"
+complete -c kw -n "__fish_kw_no_commands" -a "codestyle c" -d "Apply checkpatch on directory or file"
+complete -c kw -n "__fish_kw_no_commands" -a "maintainers m" -d "Return the maintainers and the mailing list"
+complete -c kw -n "__fish_kw_no_commands" -a "explore e" -d "Search for expression on git log or directory"
+complete -c kw -n "__fish_kw_no_commands" -a "help h" -d "Display this help mesage"
```

Comment:
These commands look extremely similar which is not good for code maintenance. How about that:
1. Create a bash hash, with the command and the description
2. Create a `for` loop and iterate over the hash and build the command

With this approach, it is going to be easy to expand the fish competition in the future.

Reply:
I totally agree with you, but, sadly, fish script doesn't have hashes natively, and I didn't find a way to create matrices in that language... I can create two arrays, one with the commands, and the other with the descriptions, but I don't think it would be clear...

Reply:
arrrgh... I've to admit that I really dislike this in the fish shell... anyway, just go ahead. We don't want to change the world, just add support for fish shell :P